src/components/SearchModal/sorting.ts

```diff
@@ -39,14 +42,17 @@ function getTokenComparator(balances: {
 }
 
 export function useTokenComparator(inverted: boolean): (tokenA: Token, tokenB: Token) => number {
+  // todo namgold: optimize this
   const balances = useAllTokenBalances()
-  // eslint-disable-next-line react-hooks/exhaustive-deps
-  const comparator = useMemo(() => getTokenComparator(balances ?? {}), [JSON.stringify(balances)])
-  return useMemo(() => {
+  const debouncedBalances = useDebounce(balances, 500)
+  const comparator = useMemo(() => getTokenComparator(debouncedBalances ?? {}), [debouncedBalances])
+  const result = useMemo(() => {
     if (inverted) {
       return (tokenA: Token, tokenB: Token) => comparator(tokenA, tokenB) * -1
     } else {
       return comparator
     }
   }, [inverted, comparator])
+  useDebug({ title: 'useTokenComparator', inverted, balances, debouncedBalances, comparator, result })
```

Comment:
Maybe only in development?

Reply:
yes. I left 2 hooks with todo which is not fully optimized yet. You guy can have a look and help me with a solutioin. If not I will remove todo and useDebug here